components/receive-amount.tsx

```diff
@@ -15,121 +26,105 @@ const satsFormatter = new Intl.NumberFormat("en-US", {
   maximumFractionDigits: 0,
 })
 
-const parseQueryAmount = (queryObject: Record<string, string>) => {
-  return {
-    amount: Number(queryObject?.amount) || 0,
-    currency: queryObject?.currency?.toUpperCase() || "USD",
+const LN_INVOICE_CREATE_ON_BEHALF_OF_RECIPIENT = gql`
+  mutation lnInvoiceCreateOnBehalfOfRecipient($walletId: WalletId!, $amount: SatAmount!) {
+    mutationData: lnInvoiceCreateOnBehalfOfRecipient(
+      input: { recipientWalletId: $walletId, amount: $amount }
+    ) {
+      errors {
+        message
+      }
+      invoice {
+        paymentRequest
+      }
+    }
   }
-}
+`
+
+const INVOICE_STALE_CHECK_INTERVAL = 5 * 60 * 1000
+const INVOICE_EXPIRE_INTERVAL = 60 * 60 * 1000
+
+export default function ReceiveAmount({ userWalletId }: { userWalletId: string }) {
+  console.log("rendering receive-amount component")
 
-export default function ReceiveAmount({
-  userWalletId,
-  updateURLAmount,
-}: {
-  userWalletId: string
-  updateURLAmount: ({ amount, currency }: { amount: number; currency: string }) => void
-}) {
+  const router = useRouter()
   const { satsToUsd, usdToSats } = useSatPrice()
-  const [satsForInvoice, setSatsForInvoice] = React.useState(0)
+  const { amount, currency } = parseQueryAmount(router.query) // USD or SATs
 
-  const urlSearchParams = new URLSearchParams(window.location.search)
-  const params = Object.fromEntries(urlSearchParams.entries())
-  const { amount, currency } = parseQueryAmount(params) // USD or SATs
+  const [regenerateCounter, setRegenerateCounter] = useState(1)
```

Comment:
Why introduce an extra state to drive invoice generation? The invoice depends on number of sats, that should be the only thing that drives the action

Reply:
This is only for triggering the invoice generation from the button when it expires.

Reply:
`satsForInvoice` has to remain a local state and not something derived from URL. If it's derived, the invoice will regenerate on every price change. 

The buttons should update satsForInvoice, and that should trigger an invoice generation. Please remove this state.

Reply:
You're right. Done.